### ISSUE TITLE
move away from CMR tokens in favor of EDL tokens

### ISSUE THREAD

**Issue by mike-gangl** [podaac/data-subscriber#98]
**ECHO-Token Deprecation Notice**
CMR Legacy Services' ECHO tokens will be deprecated soon. Please use EDL tokens and send them with the Authorization header. This document contains many mentions of ECHO-Tokens, which will soon be out of date. Instructions on how to generate an EDL token are [here](https://wiki.earthdata.nasa.gov/display/EL/How+to+Generate+a+User+Token)

we will need to update the subscriber code base to either accept a EDL token or go and generate one. Generate one fits most seamlessly with the current way of working.

**Comment by mike-gangl** [comment]
From Alica Aleman (email):

```
For Search:
If you are using Launchpad tokens, no action is needed. 
If you are using EDL tokens with the 'Authorization: Bearer' header, no action is needed. 
If you are using EDL tokens with the 'Echo-Token' header, you will need to transition to use the 'Authorization: Bearer' header. 
To use the 'Authorization: Bearer' header for CMR search, see [https://cmr.earthdata.nasa.gov/search/site/docs/search/api.html](https://urldefense.us/v3/__https://cmr.earthdata.nasa.gov/search/site/docs/search/api.html__;!!PvBDto6Hs4WbVuu7!fy94XGRgDPIPh_gGDYe_HWePbv2BEYPfL4MMiAX0d3C7gwsatvV5hGffNofGB8iWUcXbvOLXNg$). 
To use the 'Authorization: Bearer' header for CMR Access Control, see [https://cmr.earthdata.nasa.gov/ingest/site/docs/access-control/api.html](https://urldefense.us/v3/__https://cmr.earthdata.nasa.gov/ingest/site/docs/access-control/api.html__;!!PvBDto6Hs4WbVuu7!fy94XGRgDPIPh_gGDYe_HWePbv2BEYPfL4MMiAX0d3C7gwsatvV5hGffNofGB8iWUcVhMwBCMg$). 
If you are using ECHO tokens, you will need to transition to use EDL tokens.
To generate an EDL token, please visit [How to Generate a User Token](https://urldefense.us/v3/__https://wiki.earthdata.nasa.gov/display/EL/How*to*Generate*a*User*Token__;KysrKys!!PvBDto6Hs4WbVuu7!fy94XGRgDPIPh_gGDYe_HWePbv2BEYPfL4MMiAX0d3C7gwsatvV5hGffNofGB8iWUcXiWDF80g$). 
To see EDL token creation and usage examples, please visit the [CMR Data Partner User Guide - Create a Token](https://urldefense.us/v3/__https://wiki.earthdata.nasa.gov/display/CMR/CMR*Data*Partner*User*Guide*CMRDataPartnerUserGuide-ToCreateaToken__;KysrKyM!!PvBDto6Hs4WbVuu7!fy94XGRgDPIPh_gGDYe_HWePbv2BEYPfL4MMiAX0d3C7gwsatvV5hGffNofGB8iWUcUkkaOQ5w$)
```

**Comment by mike-gangl** [comment]
Ok, the token has changed a bit since we started writing this code. The end result, PR coming soon, is that the get_token(url) command will check for existing EDL tokens to use. If it finds one, it uses it. If it doens't, it creates one. it no longer deletes tokens, leaving them for re-use. We decided on the previous step such that a generated token, in use by a user for some other reason, should not be delete by us.

Currently there is no way to determine if a token was created by subscriber or some other mechanism, so we don't want to  delete it. Because the token lasts ~2 months, this isn't a big deal for us.

**Comment by edshred2000** [comment]
Thanks.

From: mike-gangl ***@***.***>
Reply-To: podaac/data-subscriber ***@***.***>
Date: Tuesday, September 27, 2022 at 8:50 AM
To: podaac/data-subscriber ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [EXTERNAL] Re: [podaac/data-subscriber] move away from CMR tokens in favor of EDL tokens (Issue #98)


From Alica Aleman (email):

For Search:

If you are using Launchpad tokens, no action is needed.

If you are using EDL tokens with the 'Authorization: Bearer' header, no action is needed.

If you are using EDL tokens with the 'Echo-Token' header, you will need to transition to use the 'Authorization: Bearer' header.

To use the 'Authorization: Bearer' header for CMR search, see [https://cmr.earthdata.nasa.gov/search/site/docs/search/api.html](https://urldefense.us/v3/__https://cmr.earthdata.nasa.gov/search/site/docs/search/api.html__;!!PvBDto6Hs4WbVuu7!fy94XGRgDPIPh_gGDYe_HWePbv2BEYPfL4MMiAX0d3C7gwsatvV5hGffNofGB8iWUcXbvOLXNg$).

To use the 'Authorization: Bearer' header for CMR Access Control, see [https://cmr.earthdata.nasa.gov/ingest/site/docs/access-control/api.html](https://urldefense.us/v3/__https://cmr.earthdata.nasa.gov/ingest/site/docs/access-control/api.html__;!!PvBDto6Hs4WbVuu7!fy94XGRgDPIPh_gGDYe_HWePbv2BEYPfL4MMiAX0d3C7gwsatvV5hGffNofGB8iWUcVhMwBCMg$).

If you are using ECHO tokens, you will need to transition to use EDL tokens.

To generate an EDL token, please visit [How to Generate a User Token](https://urldefense.us/v3/__https://wiki.earthdata.nasa.gov/display/EL/How*to*Generate*a*User*Token__;KysrKys!!PvBDto6Hs4WbVuu7!fy94XGRgDPIPh_gGDYe_HWePbv2BEYPfL4MMiAX0d3C7gwsatvV5hGffNofGB8iWUcXiWDF80g$).

To see EDL token creation and usage examples, please visit the [CMR Data Partner User Guide - Create a Token](https://urldefense.us/v3/__https://wiki.earthdata.nasa.gov/display/CMR/CMR*Data*Partner*User*Guide*CMRDataPartnerUserGuide-ToCreateaToken__;KysrKyM!!PvBDto6Hs4WbVuu7!fy94XGRgDPIPh_gGDYe_HWePbv2BEYPfL4MMiAX0d3C7gwsatvV5hGffNofGB8iWUcUkkaOQ5w$)

**Comment by mike-gangl** [comment]
Closed with release of 1.12.0
